src/main/java/uk/ac/ebi/ena/webin/cli/service/CompleteSampleService.java

```diff
@@ -0,0 +1,209 @@
+     */
+    private Sample getSampleFromXmlAPI(String sampleId, boolean test) {
+
+        RestTemplate restTemplate = getRestTemplate(sampleId);
+        HttpHeaders headers = getAuthTokenHeader();
+        ResponseEntity<String> response = executeHttpGet(restTemplate, "samples/{id}", headers, sampleId, getTest());
+        String sampleXml=response.getBody();
+        Sample sample = new Sample();
+        try {
+            DocumentBuilderFactory dbf = DocumentBuilderFactory.newInstance();
+            DocumentBuilder db = dbf.newDocumentBuilder();
+            Document doc = db.parse(new InputSource(new StringReader(sampleXml)));
+            doc.getDocumentElement().normalize();
+
+            // name
+            NodeList sampleList = doc.getElementsByTagName("SAMPLE");
+            for (int i = 0; i < Math.min(1, sampleList.getLength()); i++) {
+                Node node = sampleList.item(i);
+                if (node.getNodeType() == Node.ELEMENT_NODE) {
+                    Element element = (Element) node;
+                    sample.setName(element.getAttribute("alias"));
+                }
+            }
+
+            // taxid, organism
+            NodeList sampleNameList = doc.getElementsByTagName("SAMPLE_NAME");
+            for (int i = 0; i < Math.min(1, sampleNameList.getLength()); i++) {
+                Node node = sampleNameList.item(i);
+                if (node.getNodeType() == Node.ELEMENT_NODE) {
+                    Element element = (Element) node;
+                    if (element.getElementsByTagName("TAXON_ID").getLength() > 0) {
+                        sample.setTaxId(Integer.valueOf(element.getElementsByTagName("TAXON_ID").item(0).getTextContent()));
+                    }
+                    if (element.getElementsByTagName("SCIENTIFIC_NAME").getLength() > 0) {
+                        sample.setOrganism(element.getElementsByTagName("SCIENTIFIC_NAME").item(0).getTextContent());
+                    }
+                }
+            }
+
+            // attributes
+            NodeList attributeList = doc.getElementsByTagName("SAMPLE_ATTRIBUTE");
+            for (int i = 0; i < attributeList.getLength(); i++) {
+                Node node = attributeList.item(i);
+                if (node.getNodeType() == Node.ELEMENT_NODE) {
+                    Element element = (Element) node;
+                    String tag = element.getElementsByTagName("TAG").item(0).getTextContent();
+                    String value = element.getElementsByTagName("VALUE").getLength() > 0 ? element.getElementsByTagName("VALUE").item(0).getTextContent() : null;
+                    String units = element.getElementsByTagName("UNITS").getLength() > 0 ? element.getElementsByTagName("UNITS").item(0).getTextContent() : null;
+                    sample.addAttribute(new Attribute(tag, value, units));
+                }
+            }
+
+            return sample;
+        } catch (Exception ex) {
+            throw ServiceiException.userError(

```

Comment:
Think about the exceptions and how we should preserve the current error handling/reporting in Webin-CLI when we move its code from there to here.